### PR TITLE
Remove the "missing appenders for logger" warning when running tests with Tycho

### PR DESCRIPTION
**Pull request by erogleva**
Currently running the tests in a freshly generated Xtext project with Maven and Tycho produces the following warning:

```
log4j:WARN No appenders could be found for logger (org.eclipse.xtext.parser.antlr.AbstractInternalAntlrParser).
log4j:WARN Please initialize the log4j system properly.
```

I saw that this could be solved by adding a dependency to `org.eclipse.logging` in the configuration of the `tycho-surefire-plugin`. Would it be okay to add this to the project template? 

If another solution for this problem would be more suitable, I can adapt my PR.

### CODE REVIEW

File: org.eclipse.xtext.tests/testdata/wizard-expectations/org.xtext.example.lsMavenTychoFatjar/org.xtext.example.lsMavenTychoFatjar.parent/pom.xml [hunk 1]
```diff
@@ -232,6 +232,14 @@
 						<argLine>${platformSystemProperties} ${systemProperties} ${moduleProperties} ${additionalTestArguments}</argLine>
 						<failIfNoTests>false</failIfNoTests>
 						<useUIHarness>false</useUIHarness>
+						<dependencies>
+							<dependency>
+								<groupId>org.eclipse.xtext</groupId>
+								<artifactId>org.eclipse.xtext.logging</artifactId>
```
Comment (cdietrich) [inline org.eclipse.xtext.tests/testdata/wizard-expectations/org.xtext.example.lsMavenTychoFatjar/org.xtext.example.lsMavenTychoFatjar.parent/pom.xml]: this version looks bogus to me

**Comment by cdietrich** [comment]
i have no idea if this is a good idea. shouldnt we have xtext.logging in the target platform then?

File: org.eclipse.xtext.tests/testdata/wizard-expectations/org.xtext.example.lsMavenTychoFatjar/org.xtext.example.lsMavenTychoFatjar.parent/pom.xml (hunk 1 above)
Comment (cdietrich) [reply org.eclipse.xtext.tests/testdata/wizard-expectations/org.xtext.example.lsMavenTychoFatjar/org.xtext.example.lsMavenTychoFatjar.parent/pom.xml]: have no idea why we number it that strange

**Comment by cdietrich** [comment]
the domain model example in xtext-eclipse shows the same problem

**Comment by cdietrich** [comment]
https://wiki.eclipse.org/Tycho/Packaging_Types#eclipse-test-plugin indicates that this maybe should be a extra requirement

**Comment by erogleva** [comment]
> https://wiki.eclipse.org/Tycho/Packaging_Types#eclipse-test-plugin indicates that this maybe should be a extra requirement

Oh I see, now I noticed that here https://www.eclipse.org/tycho/sitedocs/tycho-surefire/tycho-surefire-plugin/test-mojo.html#dependencies this is also recommended, I can change this

**Comment by erogleva** [comment]
Listed now as an external requirement

**Comment by cdietrich** [comment]
@kthoms what do you think?

**Comment by cdietrich** [comment]
ping @kthoms

**Comment by erogleva** [comment]
Now I have listed it in the parent's pluginManagement

File: org.eclipse.xtext.tests/testdata/wizard-expectations/org.xtext.example.mavenTycho/org.xtext.example.mavenTycho.parent/org.xtext.example.mavenTycho.tests/pom.xml [hunk 2]
```diff
@@ -15,6 +15,10 @@
 				<groupId>org.eclipse.xtend</groupId>
 				<artifactId>xtend-maven-plugin</artifactId>
 			</plugin>
+			<plugin>
+				<groupId>org.eclipse.tycho</groupId>
+				<artifactId>target-platform-configuration</artifactId>
+			</plugin>
```
Comment (kthoms) [inline org.eclipse.xtext.tests/testdata/wizard-expectations/org.xtext.example.mavenTycho/org.xtext.example.mavenTycho.parent/org.xtext.example.mavenTycho.tests/pom.xml]: This should be obsolete, the target-platform-configuration is implicit

File: org.eclipse.xtext.tests/testdata/wizard-expectations/org.xtext.example.mavenTychoJ9/org.xtext.example.mavenTychoJ9.parent/org.xtext.example.mavenTychoJ9.tests/pom.xml [hunk 3]
```diff
@@ -15,6 +15,10 @@
 				<groupId>org.eclipse.xtend</groupId>
 				<artifactId>xtend-maven-plugin</artifactId>
 			</plugin>
+			<plugin>
+				<groupId>org.eclipse.tycho</groupId>
+				<artifactId>target-platform-configuration</artifactId>
+			</plugin>
```
Comment (kthoms) [inline org.eclipse.xtext.tests/testdata/wizard-expectations/org.xtext.example.mavenTychoJ9/org.xtext.example.mavenTychoJ9.parent/org.xtext.example.mavenTychoJ9.tests/pom.xml]: obsolete

File: org.eclipse.xtext.tests/testdata/wizard-expectations/org.xtext.example.mavenTychoP2/org.xtext.example.mavenTychoP2.parent/org.xtext.example.mavenTychoP2.tests/pom.xml [hunk 4]
```diff
@@ -15,6 +15,10 @@
 				<groupId>org.eclipse.xtend</groupId>
 				<artifactId>xtend-maven-plugin</artifactId>
 			</plugin>
+			<plugin>
+				<groupId>org.eclipse.tycho</groupId>
+				<artifactId>target-platform-configuration</artifactId>
+			</plugin>
```
Comment (kthoms) [inline org.eclipse.xtext.tests/testdata/wizard-expectations/org.xtext.example.mavenTychoP2/org.xtext.example.mavenTychoP2.parent/org.xtext.example.mavenTychoP2.tests/pom.xml]: obsolete

File: org.eclipse.xtext.xtext.wizard/src/org/eclipse/xtext/xtext/wizard/TestProjectDescriptor.xtend [hunk 5]
```diff
@@ -95,32 +95,19 @@ abstract class TestProjectDescriptor extends ProjectDescriptor {
...
+							�IF needsUiHarness�
 							<plugin>
 								<groupId>org.eclipse.tycho</groupId>
 								<artifactId>tycho-surefire-plugin</artifactId>
 								<configuration>
 									<useUIHarness>true</useUIHarness>
 								</configuration>
 							</plugin>
+							�ENDIF�
 							<plugin>
```
Comment (kthoms) [inline org.eclipse.xtext.xtext.wizard/src/org/eclipse/xtext/xtext/wizard/TestProjectDescriptor.xtend]: remove the plugin here

File: org.eclipse.xtext.tests/testdata/wizard-expectations/org.xtext.example.mavenTycho/org.xtext.example.mavenTycho.parent/org.xtext.example.mavenTycho.tests/pom.xml (hunk 2 above)
Comment (erogleva) [reply org.eclipse.xtext.tests/testdata/wizard-expectations/org.xtext.example.mavenTycho/org.xtext.example.mavenTycho.parent/org.xtext.example.mavenTycho.tests/pom.xml]: Removed

File: org.eclipse.xtext.tests/testdata/wizard-expectations/org.xtext.example.mavenTychoJ9/org.xtext.example.mavenTychoJ9.parent/org.xtext.example.mavenTychoJ9.tests/pom.xml (hunk 3 above)
Comment (erogleva) [reply org.eclipse.xtext.tests/testdata/wizard-expectations/org.xtext.example.mavenTychoJ9/org.xtext.example.mavenTychoJ9.parent/org.xtext.example.mavenTychoJ9.tests/pom.xml]: removed

File: org.eclipse.xtext.tests/testdata/wizard-expectations/org.xtext.example.mavenTychoP2/org.xtext.example.mavenTychoP2.parent/org.xtext.example.mavenTychoP2.tests/pom.xml (hunk 4 above)
Comment (erogleva) [reply org.eclipse.xtext.tests/testdata/wizard-expectations/org.xtext.example.mavenTychoP2/org.xtext.example.mavenTychoP2.parent/org.xtext.example.mavenTychoP2.tests/pom.xml]: removed

File: org.eclipse.xtext.xtext.wizard/src/org/eclipse/xtext/xtext/wizard/TestProjectDescriptor.xtend (hunk 5 above)
Comment (erogleva) [reply org.eclipse.xtext.xtext.wizard/src/org/eclipse/xtext/xtext/wizard/TestProjectDescriptor.xtend]: removed

**Comment by erogleva** [comment]
> Could you check that removal of the remaining empty target-platform-configuration plugin config still works? And then remove those entries.

Yes, thanks, it works. I have removed the entries

**Comment by erogleva** [comment]
Thank you very much for the reviews too!

**Comment by cdietrich** [comment]
i will do another pr.
having the configuration twice is not needed

**Pull request merged**